### PR TITLE
fix: support ssh format download template url

### PR DESCRIPTION
**Pull request by yugasun**
## What did you implement

Support ssh format template url, like `git@example.com/sample-service.git`, actually, many company has their private gitlab, so I think supporting these format template urls is a must.

## How can we verify it

I have add unit test for it:

```js
it('should download the service based on a regular .git URL start with git@', () => {
  const url = 'git@example.com/sample-service.git';

  return expect(downloadTemplateFromRepo(url)).to.be.fulfilled.then(() => {
    expect(spawnStub.calledOnce).to.equal(true);
    expect(downloadStub.calledOnce).to.equal(false);
    expect(spawnStub.args[0][0]).to.equal('git');
    expect(spawnStub.args[0][1][0]).to.equal('clone');
    expect(spawnStub.args[0][1][1]).to.equal(url);
  });
});

it('should download and rename the service based on a regular .git URL start with git@', () => {
  const url = 'git@example.com/sample-service.git';
  const name = 'new-service-name';

  spawnStub.resolves({
    then: callback => {
      const slsYml = path.join(process.cwd(), 'new-service-name', 'serverless.yml');
      writeFileSync(slsYml, 'service: sample-service');
      callback();
    },
  });

  return expect(downloadTemplateFromRepo(url, name)).to.be.fulfilled.then(serviceName => {
    expect(spawnStub.calledOnce).to.equal(true);
    expect(downloadStub.calledOnce).to.equal(false);
    expect(spawnStub.args[0][0]).to.equal('git');
    expect(spawnStub.args[0][1][0]).to.equal('clone');
    expect(spawnStub.args[0][1][1]).to.equal(url);
    const yml = readFileSync(path.join(newServicePath, 'serverless.yml'));
    expect(yml.service).to.equal(name);
    expect(serviceName).to.equal('sample-service');
  });
});
```

## Todos

<details>
<summary>Useful Scripts</summary>

- `npm run test:ci` --> Run all validation checks on proposed changes
- `npm run lint:updated` --> Lint all the updated files
- `npm run lint:fix` --> Automatically fix lint problems (if possible)
- `npm run prettier-check:updated` --> Check if updated files adhere to Prettier config
- `npm run prettify:updated` --> Prettify all the updated files

</details>

- [x] Write and run all tests
- [ ] Write documentation
- [x] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by hkbarton** [comment]
@medikoo could you review this?

File: lib/utils/downloadTemplateFromRepo.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ function validateUrl({ url, hostname, service, owner, repo }) {
  * @param {String} url
  */
 function isPlainGitURL(url) {
-  return url.startsWith('https') && url.endsWith('.git');
+  return (url.startsWith('https') || url.startsWith('git@')) && url.endsWith('.git');
```
Comment (medikoo) [inline lib/utils/downloadTemplateFromRepo.js:55]: This url is then provided to [`download`](https://www.npmjs.com/package/download) module.

Is it confirmed that it supports urls starting with `git@` ?

I've just tried it locally, and it crashed for me with `Error: Basic authentication must be done with the `auth` option` error

Comment (yugasun) [reply lib/utils/downloadTemplateFromRepo.js:55]: @medikoo If you want to use this method, you should add your SSH key to your private `gitlab` setting.

Comment (medikoo) [reply lib/utils/downloadTemplateFromRepo.js:55]: I wonder is it good idea then to allow such urls. It requires an extra effort from user, which may not be obvious upfront, and error message is not meaningful.

Do `https://..` urls not work with gitlab? Why do you feel it's important to support `git@..` ?

Comment (yugasun) [reply lib/utils/downloadTemplateFromRepo.js:55]: For my part, I have many private repos only can cloned by `git@xxx` format url, and I want to using this private project for serverless template. And if somebody want to use `git@xxx` url, they should know that they have ssh access right.

Comment (medikoo) [reply lib/utils/downloadTemplateFromRepo.js:55]: @yugasun did you confirmed that this change works for you locally?

On my side it doesn't work (it crashes with `Basic authentication must be done...`), even though I can without issues clone repos via such url

Comment (yugasun) [reply lib/utils/downloadTemplateFromRepo.js:55]: @medikoo Of course, this is a demo output:

```bash
# yugasun @ YUGASUN-MB0 in ~/Desktop/Playground [14:54:57] 
$ serverless create --template-url=git@github.com:yugasun/fullstack-application-vue-demo.git
Serverless: Generating boilerplate...
Serverless: Downloading and installing "fullstack-application-vue-demo"...
Serverless: Successfully installed "fullstack-application-vue-demo" 
```

Before, you use this format url, you should ensure that you can `git clone` it.

Comment (medikoo) [reply lib/utils/downloadTemplateFromRepo.js:55]: Indeed I confirm it also works on my side

**Pull request merged**